src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -634,11 +671,32 @@ private void RenderInExistingBatch(RenderQueueEntry renderQueueEntry)
                 var disposeComponentId = _batchBuilder.ComponentDisposalQueue.Dequeue();
                 var disposeComponentState = GetRequiredComponentState(disposeComponentId);
                 Log.DisposingComponent(_logger, disposeComponentState);
-                if (!disposeComponentState.TryDisposeInBatch(_batchBuilder, out var exception))
+                if (disposeComponentState.RequiresAsyncDisposal())
                 {
-                    exceptions ??= new List<Exception>();
-                    exceptions.Add(exception);
+                    if (!disposeComponentState.TryDisposeInBatch(_batchBuilder, out var exception))
+                    {
+                        exceptions ??= new List<Exception>();
+                        exceptions.Add(exception);
+                    }
                 }
+                else
+                {
+                    var result = disposeComponentState.DisposeInBatchAsync(_batchBuilder);
+                    if (result.IsCompleted)
+                    {
+                        if (result.IsFaulted && !result.IsCanceled)
```

Comment:
Comment why canceled tasks are ignored?

Reply:
In most of our other component lifecycle methods, we regard a "cancelled" response as indicating "this is no longer relevant to me" so it's not an error, it's just something to be discarded.

For example, if during `OnInitializedAsync` you return a `Task` corresponding to some API call, but then the user navigates away before the API call completes, the task will often complete as cancelled because it was originally supplied a `CancellationToken` that fires when the component is disposed.

**However** I'm not sure the same logic applies here. Why would a disposal task ever be cancelled, other than due to an error? It's not like something happens even later than `Dispose` that would count as a valid justification for aborting the disposal. Disposal can't be meaningfully aborted.

So @javiercn should we err on the side of caution and treat cancellation as a fatal exception? This gives us the flexibility to add support for it in the future. Whereas if we ignore cancellation now, we can never change that to an error in the future.

Reply:
I've switched it back to treat cancellation s an error just to be conservative. I'm not sure when it is critical to treat as an error vs when it is not (since you can write code to capture cancellations yourself) which is what I would have expected to happen.

I can see where just ignoring it can be useful if you use it to signal something in a fire and forget context.